src/ReportManager/SchemaFormContext/SchemaForm/fields/fieldDetailsFactory/index.test.js

```diff
@@ -0,0 +1,32 @@
+import { textFieldDetailsFactory } from './index';
+
+describe('ReportManager - SchemaForm - fieldDetailsFactory', () => {
+
+  test('builds details object of text field properly', () => {
+    const schemaDetails = {
+      description: 'A hint of the field',
+      title: 'Important Text',
+      deprecated: true,
+      default: 'initial value'
+    };
+    const uiDetails = {
```

Comment:
And this one `uiSchema`. Remember that there are two things: the schema and the field details. Each one of those can be converted to the other. The "schema" is what we store in the DB and it respects the JSON schema protocol, with the UI complementing it with what we need that is not supported by the protocol. The "field details" is how we call the object we construct with the information on how to render the field. Those two terms are properly separated in the EFB. In there, the field details come from a factory and it is handled by the drag and drop interface, and the schema is generated by a hook `useSchema` from the details object. In here, we are mixing those two concepts in a few places, which will cause confusion. 
Summary:
- Schema: stored in the DB, it is built by the jsonSchema (which adders to the latest draft of JSON schema) and the uiSchema (which was defined by us complementing the json schema)
- Field details: a single object with all the information needed to render a field (type, placeholder, label, etc...)
One of those can be transformed to the other when it is convenient. The schema is useful for the backend to be stored and to run the schema validations, the field details is convenient for the frontend to render the field.